Controllers/AuthenticationController.py

```diff
@@ -24,19 +32,65 @@ def logout():
     AuthenticationService.logout(sessionId)
     return Response("User logged out", status=200)
 
+
 """
 Request params: first name, last name, email (will be used as username), password
 Return: Success or failure codes    
 """
 @auth.route("/signup", methods=["POST"])
 def signup():
-    user = {"firstName": request.form["firstName"], 
-            "lastName": request.form["lastName"], 
+    user = {"firstName": request.form["firstName"],
+            "lastName": request.form["lastName"],
             "email": request.form["email"],
             "password": request.form["password"]
             }
 
     if (not AuthenticationService.signup(user)):
-            return Response(status=400)
-    
-    return Response(status=200) 
\ No newline at end of file
+        return Response(status=400)
+
+    return Response(status=200)
+
+
+@auth.route("/forgot-password", methods=["POST"])
+def forgotPassword():
+    try:
+        userObject = User.objects.get(email=request.form["email"])
+        if userObject == None:
+            return Response("No account with given email found", status=403)
+        else:
+            session = Session(user=userObject)
+            session.save()
+            try:
+                subject = "[Agriworks] Reset password"
+                html = "<p>Hi there,</p><p>We heard you lost your password. No worries, just click the link below to reset your password.</p><p>You can safely ignore this email if you did not request a password reset</p><br/><a href=\"http://localhost:8080/reset-password/{0}\">http://localhost:8080/reset-password/{0}</a><br/><p>Thanks,</p><p>Agriworks Team</p>".format(
+                    session.sessionId)
+                msg = Message(recipients=[userObject.email],
+                              subject=subject, html=html)
+                mail.send(msg)
+                return Response(status=200)
+            except:
+                return Response("Unable to send email", status=400)
+    except:
+        return Response("Invalid header", status=400)
```

Comment:
Why would the header be invalid ?

Reply:
I will change the error codes to be more unique to each error in user feedback feature.